auth/env_keychain.go

```diff
@@ -12,27 +12,48 @@ import (
 	"github.com/pkg/errors"
 )
 
-// NewKeychain returns either a EnvKeychain or a authn.DefaultKeychain depending on whether the provided environment variable is set
-func NewKeychain(envVar string) authn.Keychain {
-	_, ok := os.LookupEnv(envVar)
-	if !ok {
-		return authn.DefaultKeychain
+// DefaultKeychain returns a keychain containing authentication configuration for the given images
+// from the following sources, if they exist, in order of precedence:
+// the provided environment variable and the docker config.json file.
+func DefaultKeychain(envVar string, images ...string) (authn.Keychain, error) {
```

Comment:
What do you think about moving the `EnvRegistryAuth` `const` into this package and using this value directly in the `DefaultKeychain` impl, instead of passing in the `envVar` parameter?

Reply:
Sounds good!